packages/zoe/src/zoeService/zoe.js

```diff
@@ -168,11 +159,9 @@ function makeZoe(vatAdminSvc) {
                 ...instanceRecord.brandKeywordRecord,
                 [keyword]: brand,
               };
-              brandToPurse.init(brand, E(issuer).makeEmptyPurse());
-              instanceTable.update(instance, {
-                issuerKeywordRecord: instanceRecord.issuerKeywordRecord,
-                brandKeywordRecord: instanceRecord.brandKeywordRecord,
-              });
+              if (!brandToPurse.has(brand)) {
+                brandToPurse.init(brand, E(issuer).makeEmptyPurse());
+              }
```

Comment:
Shouldn't the keywordRecords be updated when a new Brand is added?

Reply:
They are, a few lines above this.